tests/worksheet_test.py

```diff
@@ -914,6 +916,140 @@ def test_get_all_records_numericise_unformatted(self):
         d0 = dict(zip(rows[0], expected_values))
         self.assertEqual(read_records[0], d0)
 
+    @pytest.mark.vcr()
+    def test_get_records(self):
+        self.sheet.resize(5, 3)
+        rows = [
+            ["A1", "B1", "C1"],
+            [1, 2, 3],
+            [4, 5, 6],
+            [7, 8, 9],
+            [10, 11, 12],
+        ]
+        cell_list = self.sheet.range("A1:C5")
+        for cell, value in zip(cell_list, itertools.chain(*rows)):
+            cell.value = value
+        self.sheet.update_cells(cell_list)
+
+        # test1 - set max_row only
+        read_records = self.sheet.get_records(last_row=3)
+        d0 = dict(zip(rows[0], rows[1]))
+        d1 = dict(zip(rows[0], rows[2]))
+        records_list = [d0, d1]
+        self.assertEqual(read_records, records_list)
+
+        # test2 - set first_row only
+        read_records = self.sheet.get_records(first_row=3)
+        d0 = dict(zip(rows[0], rows[2]))
+        d1 = dict(zip(rows[0], rows[3]))
+        d2 = dict(zip(rows[0], rows[4]))
+        records_list = [d0, d1, d2]
+        self.assertEqual(read_records, records_list)
+
+        # test3 - set both max_row and first_row unequal to each other
+        read_records = self.sheet.get_records(first_row=3, last_row=4)
+        d0 = dict(zip(rows[0], rows[2]))
+        d1 = dict(zip(rows[0], rows[3]))
+        records_list = [d0, d1]
+        self.assertEqual(read_records, records_list)
+
+        # test4 - set max_row and first_row equal to each other
+        read_records = self.sheet.get_records(first_row=3, last_row=3)
+        d0 = dict(zip(rows[0], rows[2]))
+        records_list = [d0]
+        self.assertEqual(read_records, records_list)
+
+        # test5 - set head only
```

Comment:
shouldn't this test return all rows? it only looks like it returns rows[2:4] when it should be rows[2:5] (or rows[2:])

Reply:
it does return all rows (head is second row, first row is removed)